### PR TITLE
feat: add more convenience methods that are designed with intent in mind

### PR DESCRIPTION
**Pull request by JohannesDienst-askui**
Add/modify convenience methods:

* `clickText({text: string})` -> Breaking change 
* `typeIntoTextfield({textToWrite: string})`
* `clickSwitch({label: string})`
* `clickCheckbox({label: string})`
* `clickButton({})` -> Breaking change

`clickText({})` and `clickButton({})` introduce a braking change as they now take objects instead of a single `string` argument.

Open questions:
* I put `Relations` into `dsl.ts`. It is generated over at `askui-inference`, correct? How do I add this properly to the TypeScript ADK?
* Is the design intuitive enough or is there a better way than using objects as parameters?

### CODE REVIEW

File: packages/askui-nodejs/src/execution/dsl.ts
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/dsl.ts:15]: You are right, the whole `dsl.ts` file is generated through the askui-inference. @programminx-askui or @mlikasam-askui can help you out with interating it there. Alternatively, you could also simply move this to another file, e.g., `ui-control-client.ts` or a new file.

Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/dsl.ts:15]: I think we should improve on the naming as this does not contain all relations but only a subset that are relevant for a specific use case.

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-control-client.ts:400]: This can only happen if someone is going around the Typescript compiler some way from my perspective using a relation that we don't support. What about throwing an error in this case instead to let the user know so that the user is able to fix this?

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-control-client.ts:382]: Minor thing (optional) to improve code readability, maintainability etc.: I would either move switch to a dedicated function or check if `params.relation.type` is included in an array of all the values included in the union type `Relations` and if not, do whatever is under `default` else call `command[params.relation.type]().text(params.relation.text)`. Typescript may not like it but could be disabled for this specific purpose.

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-control-client.ts:457]: Minor thing (optional): I think it would help in the future if these convenience functions were implemented more consistently (with `clickButtton()`,  e.g., calling exec always at the end on the command build previously, maybe even sharing some part of the code, e.g., the relationship building part (⟶ currently, switch statement).

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-control-client.ts:603]: I think that these both correlate with each other can lead to confusing edge cases and increase complexity, e.g., what happens if `regex` and `exact` are both set to `true`. What about combining these two into one string union type, e.g., `type: 'similar' | 'exact' | 'regex' = 'similar'` or `type?: 'exact' | 'regex'`?

**Review by adi-wan-askui — CHANGES_REQUESTED** [review]
I like the switch from simple string params to objects as this gives more flexibility in the future and simulates keyword arguments of other languages, e.g., Python, that a lot of our users highly enjoy. It is unfortunate that this is a breaking change but I think it is fine as it is still quite new and not many people have probably adapted it thoroughly.

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (JohannesDienst-askui) [reply packages/askui-nodejs/src/execution/ui-control-client.ts:400]: Good idea. Implemented it this way now :)

File: packages/askui-nodejs/src/execution/dsl.ts
Comment (JohannesDienst-askui) [reply packages/askui-nodejs/src/execution/dsl.ts]: Not sure if `Anchor` is a better name for it, but we used it already in the docs and it makes kinda sense.

I am open to better suggestions.

**Comment by JohannesDienst-askui** [comment]
@adi-wan-askui I hope all comments got addressed 😃 

Thanks for the great review!

File: packages/askui-nodejs/src/execution/dsl.ts
Comment (adi-wan-askui) [reply packages/askui-nodejs/src/execution/dsl.ts]: Nice term but I think it may be ambiguous and like you said, we already use the term `Relations` extensively. As this is a subset of all possible relations, what about a name like `RelationsForConvenienceMethods` (also clunky, maybe you can think of a better prefix but the general idea is to use `Relations...` prefix and specify the subset afterwards)?

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (adi-wan-askui) [reply packages/askui-nodejs/src/execution/ui-control-client.ts]: Very nice approach :)

File: packages/askui-nodejs/src/execution/dsl.ts
Comment (JohannesDienst-askui) [reply packages/askui-nodejs/src/execution/dsl.ts]: My first programming language was Java, so clunkiness is not a problem for me 😉 .

I like your thinking there and would go with your suggestion `RelationsForConvenienceMethods`. Better clunky and expressive than not clunky and ambiguous.

**Comment by JohannesDienst-askui** [comment]
@adi-wan-askui Pushed the final edits. Waiting for your approval 😃 .

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-control-client.ts:381]: Just some minor ideas for improving the expressiveness and conciseness of the code (optional):

```suggestion
  private extendCommandWithRelationToText(
    command: FluentFiltersOrRelations,
    relation: RelationsForConvenienceMethods,
    text: string,
  ) {
    switch (relation) {
      case 'nearestTo':
        return command.nearestTo().text(text);
      case 'leftOf':
        return command.leftOf().text(text);
      case 'above':
        return command.above().text(text);
      case 'rightOf':
        return command.rightOf().text(text);
      case 'below':
        return command.below().text(text);
      case 'contains':
        return command.contains().text(text);
      default:
        throw new ValidationError(`Relation "{relation}" unknown. Please, use one of the following relations: {relationsForConvenienceMethods.map(r => `"{r}"`).join(", ")}`).;
    }
  }
```

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-control-client.ts:604]:
```suggestion
        throw new ValidationError(`Type "{params.type}" unknown. Please, use one of the following types: "similar", "exact", "regex".`);
```

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-control-client.ts:584]: What about making it optional or moving it below `relation` so you can add `'similar'` as the default?

File: packages/askui-nodejs/src/execution/ui-control-client.ts
Comment (adi-wan-askui) [inline packages/askui-nodejs/src/execution/ui-control-client.ts:573]:
```suggestion
   * @property {('similar' | 'exact' | 'regex')} params.type - Whether the text is matched using similarity,
```

**Review by adi-wan-askui — APPROVED** [review]
Nicely done :) Just found some minor optimisation potentials. Feel free to skip.

**Pull request merged**
